components/utils/metadata.js

```diff
@@ -0,0 +1,197 @@
+import { sanitizeUrl } from './url'
+
+const fetchMetadata = (path) => {
+  const reqUrl = new URL(`/api/metadata`, window.location.origin)
+
+  const params = new URLSearchParams()
+  params.append('path', path)
+  reqUrl.search = params.toString()
+
+  return fetch(reqUrl)
+}
+
+export const inspectDataset = async (url) => {
+  // fetch zmetadata to figure out compression and variables
+  const sanitized = sanitizeUrl(url)
+
+  let response
+  try {
+    response = await fetchMetadata(sanitized)
+  } catch (e) {
+    // Show generic error message when request fails before response can be inspected.
+    throw new Error(
+      'A network error occurred. This could be a CORS issue or a dropped internet connection.'
+    )
+  }
+
+  if (!response.ok) {
+    const statusText = response.statusText ?? 'Dataset request failed.'
+    if (response.status === 403) {
+      throw new Error(
+        `STATUS 403: Access forbidden. Ensure that URL is correct and that dataset is publicly accessible.`
+      )
+    } else if (response.status === 404) {
+      throw new Error(
+        `STATUS 404: ${statusText} Ensure that URL path is correct.`
+      )
+    } else {
+      throw new Error(
+        `STATUS ${response.status}: ${statusText}. URL: ${sanitized}`
+      )
+    }
+  }
```

Comment:
could we move this into a helper function  to handle the generation of error messages based on the status code?
```suggestion
 if (!response.ok) {
  const statusText = response.statusText ?? 'Dataset request failed.';
  const errorMessage = generateErrorMessage(response.status, statusText, sanitized);
  throw new Error(errorMessage);
}
```

the helper function : `generateErrorMessage()`

```javascript
function generateErrorMessage(status, statusText, sanitizedUrl) {
  switch (status) {
    case 403:
      return `STATUS 403: Access forbidden. Ensure that URL is correct and that dataset is publicly accessible.`;
    case 404:
      return `STATUS 404: ${statusText}. Ensure that URL path is correct.`;
    case 500:
    case 502:
    case 503:
      return `STATUS ${status}: ${statusText}. The server encountered an error. Please try again later.`;
    default:
      return `STATUS ${status}: ${statusText}. URL: ${sanitizedUrl}`;
  }
}
```